### ISSUE TITLE
Allow linkStyle to be animated

### ISSUE THREAD

**Issue by cmnstmntmn** [mermaid-js/mermaid#5574]
### Proposal

I'm using Github gists for documentation, and i would like to have a way to allow animation between the notes.

I was trying to add custom css, but **keyframes** are not supported.

Possible solutions:

### Current implementation - no animation
Length |	1	| 2	|3
------|---| ----|--|
Normal|---|----|-----
Normal with arrow|-->|--->|---->
Thick|===|====|=====
Thick with arrow|==>|===>|====>
Dotted|-.-|-..-	|-...-
Dotted with arrow|-.->|-..->|-...->

### Proposed implementation - with animation (preferred)

> `^` pin operator will mark that:

- a dotted line will have a moving infinite animation
 
![dotted-line](https://github.com/mermaid-js/mermaid/assets/2312755/184e31de-d66b-4ae6-9835-7aab4d136b8e)

- a normal line will have a fade in/ fade out infinite animation

![normal-line](https://github.com/mermaid-js/mermaid/assets/2312755/9ddead53-30cc-4674-b382-af4e9c9767e4)

### How to mark a line for having an animated behaviour

Length |	1	| 2	|3
------|---| ----|--|
Normal|---^|----^|-----^
Normal with arrow|-->^|--->^|---->^
Thick|===^|====^|=====^
Thick with arrow|==>^|===>^|====>^
Dotted|-.-^|-..-^|-...-^
Dotted with arrow|-.->^|-..->^|-...->^
 
## Add a new `keyframesDef` property
I think adding support for keyframes like `keyframesDef`, the same way like `classDef` 

```
graph RL
    A:::className -.-> B
    classDef className fill:green;
    keyframesDef animationName 0% stroke-dashoffset: 0, 100% stroke-dashoffset: 0 color: red
    linkStyle 0 animation: animationName 1s linear infinite;
```

### Example

https://d2lang.com/tour/style#animated

![example-nodes](https://github.com/mermaid-js/mermaid/assets/2312755/f8478b56-82a6-423d-9685-2766918cfd67)

**Comment by cmnstmntmn** [comment]
I've made some progress on this.

![animated-graph](https://github.com/mermaid-js/mermaid/assets/2312755/35e7ddaa-8fe8-45f1-bfb3-982c58ea3bc1)

**Comment by ingo-eichhorst** [comment]
Alternatively you can always apply any styling to the SVG in a HTML context. Here is an example which gives you 100% freedom.
https://gist.github.com/ingo-eichhorst/cd18efd5cbc3dc60cb6095c0affe3e65

**Comment by knsv** [comment]
Proposing a new syntax for this. The key thing is to be able to attach an id the edge. After trhat we use the same syntax for edges as we have to add additional info to a node.

Easiest case:
```
flowchart LR
        A e1@==> B
        e1@{ animate: true}
```
## Selecting type of animation
V1 will support 2, `fast` and `slow`

```
flowchart LR
  A e1@--> B
  e1@{ animation: fast}
```
This is a shorthand for `{ animate: true, animation: fast}`

## classDef statements
You can also add animations using classDef statements

```
flowchart LR
  A e1@--> B
  classDef animate stroke-dasharray: 9\,5,stroke-dashoffset: 900,animation: dash 25s linear infinite;
  class e1 animate
```
Note the escaped comma in the dasharray. This is because ',' is used as a delimiter between styles mermaid.

**Comment by TheDevelolper** [comment]
yeah but it seems to break when you try to use the class twice like so: 

```mermaid 

  Web e@==> ControlLoop
  Mobile  e@== text ==> ControlLoop
  
  e@{ animate: true, animation: slow }
```

it will only animate one but not the other. Is this expected?

**Comment by jdespatis** [comment]
I have the same behavior, any idea to avoid to clone the `e@{ animation: slow }` ?
